ct-nz Issue 2: Initial react router setup for list and add item

connietran-dev opened the pull request:
## Description

<!-- What does this code change? Why did I choose this approach? Did I learn anything worth sharing? Reminder: This will be a publicly facing representation of your work (READ: help you land that sweet dev gig). -->

We added react-router-dom and created Routes for the list view and add item view.

For now, we kept it simple and added 2 buttons that change the browser URL and renders either the List.js component or AddItem.js component.

## Related Issue

<!-- If you write "closes" followed by the Github issue number, it will automatically close the issue for you when the PR merges -->

Closes #2 

## Acceptance Criteria

<!-- Include AC from the Github issue -->

- react-router-dom has been added as a project dependency
- Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
- When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
- Whichever view is selected, the corresponding link should display in bold text


## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<!-- If UI feature, take provide screenshots -->

![image](https://user-images.githubusercontent.com/61371242/125373671-91a17380-e353-11eb-8927-c52e2e464368.png)

### After

<!-- If UI feature, take provide screenshots -->

![image](https://user-images.githubusercontent.com/61371242/125373709-a251e980-e353-11eb-8a9d-fe5f10273b8d.png)


## Testing Steps / QA Criteria

<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->

- Run `npm install` to install `react-router-dom`
- Run `npm start` to start the app on your localhost
- Click either "View List" or "Add Item" to change the browser URL and the view to the List or Add Item component

anderswift reviewed (commented):
We're off to a good start, though I'm sure you two were itching to do more just like me and @SheilaRMoore! Since this is kind of a light week, I thought I'd take the time to pass on some things I've found helpful from some of my past code reviews, and also share some things I've learned about a11y that I think would be useful here. The a11y items are the only things I feel strongly about including, and I'm open to feedback if anyone has other ideas about how to improve in that area because I know I still have a lot to learn. The rest you can take just as suggestions/conversation starters.

connietran-dev commented:
@anderswift @SheilaRMoore - Great comments from you Ander! Nick and I addressed your comments in our latest commit:

- We removed the buttons from the NavLinks and converted them into an unordered list with an aria-label for a11y.
- We changed the className of the navigation elements to be more specific for our CSS selectors. We're starting to use the BEM (Block-Element-Modifier) methodology for naming classes. We can discuss the next time we meet if you'd like an overview.
- We added a "Not Found" component for 404's.
- We changed the HTML in the AddItem and List component to <main> to be more semantic. 

Very good points made, so thank you for those! 

Please let us know if you have any questions.

anderswift reviewed (approved):
Looks fabulous! 😙👌

SheilaRMoore reviewed (commented):
I double checked here just to make sure it was latest version of all dependcies along with what we have to hopefully avoid a merge conflicts here. So far we are ok.

SheilaRMoore reviewed (commented):
I think this was a good semantic change here.

SheilaRMoore reviewed (approved):
I think you guys did a good job overall. I like the changed that were made because I feel that they maybe helpful to us as we continue to work on this project.

yenly commented on src/App.js:
```diff
@@ -1,25 +1,51 @@
+} from 'react-router-dom';
+
+// External Files
 import logo from './logo.svg';
```
As part of removing boilerplate CRA code, you can remove this logo too.

yenly commented on src/App.js:
```diff
@@ -1,25 +1,51 @@
 import logo from './logo.svg';
 import './App.css';
 
+// Components
+import List from './components/List';
+import AddItem from './components/AddItem';
+import NotFound from './components/NotFound';
```
I like how your approach in creating placeholder components for each view.

yenly reviewed (approved):
You both did a great job! 👏🏽 A gentle reminder to update your PR to reflect the changes you've incorporated as you iterate on the work. This will help your reviewers know to run `npm  install` when they pull down your branch to test and to have context on what the changes do.

Cjstoney reviewed (approved):
This PR looks really great! I think that @yenly was spot on with their notes as well. Love the import placeholders! Imports can get seriously unruly at times and organizing them the way that you have is very helpful.

abowler2 commented on src/App.css:
```diff
@@ -20,3 +20,11 @@
 .App-link {
```
Given the changes that you made in the `App` component these original classes are no longer being used (with the exception of the centering in `.App`). I would recommend removing them so that your CSS is easier to manage as the app grows.

This isn't something that necessarily needs to be done in this PR, just wanted to mention it.

abowler2 commented on src/App.js:
```diff
@@ -1,25 +1,51 @@
 import logo from './logo.svg';
 import './App.css';
 
+// Components
```
I like the organization comments/structure you went with for the imports.

abowler2 commented on src/App.js:
```diff
@@ -1,25 +1,51 @@
+          <Route component={NotFound} />
+        </Switch>
+
+        <nav className="menu" aria-label="Main Navigation">
```
aria-label = 😍

abowler2 reviewed (approved):
Really nice work!!  I left a comment re: the CSS but consider that food for thought as you progress with the app. 

![](https://media4.giphy.com/media/iJ8LTCC41PmP06wL0h/giphy.gif)
